daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/TypeCalculator.scala

```diff
@@ -125,67 +148,79 @@ abstract class TypeCalculator[A <: AnyRef, B <: AnyRef](val srcType: NodeInfo.Ki
 
   @throws(classOf[java.io.IOException])
   final private def writeObject(out: java.io.ObjectOutputStream): Unit = serializeObject(out)
+
+  protected def normalizeArg(x: DataValuePrimitive, xType: NodeInfo.Kind): DataValuePrimitive = {
```

Comment:
All of this normalizeArg stuff shouldn't be nessasary. In theory we know at compile time exactly what types we expect to see, and can specialize then. In practice, we convert all integral types to JBigInt at compile time, and so must do the same at runtime.

Reply:
Add a comment to the code stating just what you said. This should be recorded for someone else reading the code.